src/database/queries/connection.js

```diff
@@ -0,0 +1,205 @@
+				where: {
+					[Op.or]: [
+						{ user_id: userId, friend_id: friendId },
+						{ user_id: friendId, friend_id: userId },
+					],
+					status: common.CONNECTIONS_STATUS.REQUESTED,
+					created_by: friendId,
+				},
+				individualHooks: true,
+				transaction: t,
+			})
+			if (deletedCount != 2) {
+				throw new Error('Error while deleting from "ConnectionRequest"')
+			}
+
+			const friendRequestData = [
+				{
+					user_id: userId,
+					friend_id: friendId,
+					status: common.CONNECTIONS_STATUS.ACCEPTED,
+					created_by: friendId,
+					updated_by: userId,
+					meta,
+				},
+				{
+					user_id: friendId,
+					friend_id: userId,
+					status: common.CONNECTIONS_STATUS.ACCEPTED,
+					created_by: friendId,
+					updated_by: userId,
+					meta,
+				},
+			]
+
+			const requests = await Connection.bulkCreate(friendRequestData, {
+				transaction: t,
+			})
+
+			return requests
+		})
+
+		return requests
+	} catch (error) {
+		throw error
+	}
+}
+
+exports.rejectRequest = async (userId, friendId) => {
+	try {
+		const updateData = {
+			status: common.CONNECTIONS_STATUS.REJECTED,
+			updated_by: userId,
+			deleted_at: Date.now(),
+		}
+
+		return await ConnectionRequest.update(updateData, {
+			where: {
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				created_by: friendId,
+			},
+			individualHooks: true,
+		})
+	} catch (error) {
+		return error
+	}
+}
+exports.findOneRequest = async (userId, friendId) => {
+	try {
+		const connectionRequest = await ConnectionRequest.findOne({
+			where: {
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+				created_by: friendId,
+			},
+			raw: true,
+		})
+
+		return connectionRequest
+	} catch (error) {
+		throw error
+	}
+}
+
+exports.checkPendingRequest = async (userId, friendId) => {
+	try {
+		const result = await Connection.findOne({
+			where: {
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+			},
+			raw: true,
+		})
+		return result
+	} catch (error) {
+		return error
+	}
+}
+
+exports.getSentAndReceivedRequests = async (userId) => {
+	try {
+		const result = await Connection.findAll({
+			where: {
+				[Op.or]: [{ user_id: userId }, { friend_id: userId }],
+				status: common.CONNECTIONS_STATUS.REQUESTED,
+			},
+			raw: true,
+		})
+		return result
+	} catch (error) {
+		return error
+	}
+}
+
+exports.getConnection = async (userId, friendId) => {
+	try {
+		const result = await Connection.findOne({
+			where: {
+				[Op.or]: [
+					{ user_id: userId, friend_id: friendId },
+					{ user_id: friendId, friend_id: userId },
+				],
+				status: common.CONNECTIONS_STATUS.ACCEPTED,
```

Comment:
@nevil-mathew if status is blocked then it will give false, and they can re-initiate the connection.

Reply:
We don't have a block flow as of now.

Reply:
added block check for initiate api